src/Map/Map.tsx

```diff
@@ -102,32 +86,69 @@ const Map: React.FC<Props> = (props) => {
       countFontColor,
     },
     pointOptions: { pointColor, pointRadius, onPointClick, onClickZoomIn },
-  } = applyDefaults(props)
+  } = applyLayerDefaults(props)
+
+  const sourceJson = props.sourceJson
+  const height = props.initialState?.height || '800px'
+  const width = props.initialState?.width || '800px'
 
-  mapboxgl.accessToken = token
+  mapboxgl.accessToken = props.token
 
+  // initialize map
   useEffect(() => {
-    if (mapRef.current) return undefined // initialize map only once
+    if (mapRef.current) return undefined
+
+    // defaults
+    const long = props.initialState?.long || -3.5
+    const lat = props.initialState?.lat || 55
+    const zoom = props.initialState?.zoom || 5
+    const style =
+      props.initialState?.style || 'mapbox://styles/mapbox/light-v11'
+
     mapRef.current = new mapboxgl.Map({
       container: mapContainer.current!,
       style: style,
       center: [long, lat],
       zoom: zoom,
       attributionControl: false,
     })
-  })
 
+    // clean up
+    return () => {
+      mapRef?.current?.remove()
+      mapRef.current = null
+    }
+  }, [
+    props.initialState?.style,
+    props.initialState?.long,
+    props.initialState?.lat,
+    props.initialState?.zoom,
```

Comment:
yea it does get bigger, but it's ok we can abstract later when we have a landing page:)